### ISSUE TITLE
[Reactive] Support R2DBC in entity generator

### ISSUE THREAD

**Issue by cbornet** [jhipster/generator-jhipster#11542]
We now have support for r2dbc in the main generator and entity generation for NoSQL databases.
So remains r2dbc entity generation.

Some things that make it not so easy:
* There's currently no support for query derivation although that should arrive soon : https://github.com/spring-projects/spring-data-r2dbc/issues/282
* r2dbc is not an ORM so the relational code has to be handled explicitly. Maybe the first version could be done without support for relationships.

**Comment by mraible** [comment]
I've added a bug bounty as this seems like a large task that will be difficult to do.

**Comment by gzsombor** [comment]
I would like to work on it - I've already found a couple of issue, which prevented running the reactive app, and login to it.

**Comment by mraible** [comment]
Go for it @gzsombor!

**Comment by brahimk** [comment]
Hello @gzsombor 

Do you want help on this issue ?

thanks.

**Comment by gzsombor** [comment]
@brahimk thanks for asking, there are some progress implementing this - and there is an open PR about
https://github.com/jhipster/generator-jhipster/pull/11742 . 
Currently I'm blocked by three things:
* Spring-boot 2.3 support is not yet merged, so building and running the CI is not possible out of the box. I hope, this can be solved with using the jitpack.io repository.
* Spring-data-r2dbc bug - https://github.com/spring-projects/spring-data-r2dbc/pull/390 - it's not yet merged, and released, unfortunately this error is triggered by our CI suite.
* IRL problems - like lack of time/travel/holiday

**Comment by mraible** [comment]
@gzsombor Are you still blocked? Can we split this up into smaller issue and give you more $$ to make it happen?

**Comment by murdos** [comment]
@mraible : as far as I can tell, this ticket should be closed as resolved, since https://github.com/jhipster/generator-jhipster/pull/11742 has been merged.

**Comment by mraible** [comment]
Fixed by https://github.com/jhipster/generator-jhipster/pull/11742. I'm not sure how to do the bug bounty payout. Maybe @pascalgrimaud can help?

**Comment by gzsombor** [comment]
Yes, sorry, the code has been merged, but there are a couple of open issues, in decreasing importance:
* filtering is not implemented
* the generated code is still more complex, and need more maintenance, then I would like to have. If someone adds a new column to an entity - they need to change a couple of things: add it into the mapper class, the repository impl class, and to the sql helper class. As these code looks pretty boring and repetitive, it would be nice, if this could be created either compile time (with an annotation processing tool, with something similar to mapstruct), or runtime.
* Doesnt work well with MSSQL and Oracle.

I guess, they can be implemented - and improved - if there will be user demand :wink:

**Comment by pascalgrimaud** [comment]
@mraible : as far as I can understand, the work was done mainly by @gzsombor with the 86 commits so the full bounty is for him, and well deserved
if other people participated, just tell us, and we'll see what we can do

**Comment by pascalgrimaud** [comment]
@gzsombor : approved
